pkg/server/api/entry/v1/service.go

```diff
@@ -80,7 +80,61 @@ func (s *Service) BatchUpdateEntry(ctx context.Context, req *entry.BatchUpdateEn
 }
 
 func (s *Service) BatchDeleteEntry(ctx context.Context, req *entry.BatchDeleteEntryRequest) (*entry.BatchDeleteEntryResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "method BatchDeleteEntry not implemented")
+	var results []*entry.BatchDeleteEntryResponse_Result
+	for _, id := range req.Ids {
+		results = append(results, s.deleteEntry(ctx, id))
+	}
+
+	return &entry.BatchDeleteEntryResponse{
+		Results: results,
+	}, nil
+}
+
+func (s *Service) deleteEntry(ctx context.Context, id string) *entry.BatchDeleteEntryResponse_Result {
+	log := rpccontext.Logger(ctx)
+
+	if len(id) == 0 {
+		log.Error("Invalid request: missing entry ID")
+		return &entry.BatchDeleteEntryResponse_Result{
+			Id:     id,
+			Status: api.CreateStatus(codes.InvalidArgument, "missing entry ID"),
+		}
+	}
+
+	log = log.WithField(telemetry.RegistrationID, id)
+
+	cEntry, err := s.ds.FetchRegistrationEntry(ctx, &datastore.FetchRegistrationEntryRequest{
+		EntryId: id,
+	})
+	if err != nil {
+		log.WithError(err).Error("Failed to fetch entry")
+		return &entry.BatchDeleteEntryResponse_Result{
+			Id:     id,
+			Status: api.CreateStatus(codes.Internal, "failed to fetch entry: %v", err),
+		}
+	}
+	if cEntry.Entry == nil {
+		log.Error("Invalid request: entry not found")
+		return &entry.BatchDeleteEntryResponse_Result{
+			Id:     id,
+			Status: api.CreateStatus(codes.NotFound, "entry not found"),
+		}
+	}
```

Comment:
The datastore is already supposed to handle failing if the entry does not already exist but I think it is broken and returns a code.Unknown status. We should fix the datastore to return codes.NotFound if the entry doesn't exist.